.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,58 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Checkstyle, Findbugs, Doc Check, etc.
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  pull_request:
+    branches: [ github_actions, master ]
```

Comment:
are wildcards allowed for branches? we will need these to run on release branches (ex. `branch-2.x`) as well